config/src/config/safety_rules_config.rs

```diff
@@ -52,5 +56,15 @@ pub enum SafetyRulesService {
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
 #[serde(deny_unknown_fields)]
 pub struct RemoteService {
-    pub server_address: SocketAddr,
+    pub server_address: NetworkAddress,
+}
+
+impl RemoteService {
+    pub fn server_address(&self) -> SocketAddr {
+        self.server_address
+            .to_socket_addrs()
+            .expect("server_address invalid")
```

Comment:
You'll also get the internal error message and stack trace. So for the first, it would be something to the extent that it wasn't a valid representation, whereas the next one would be that there no known hosts listening at that service. i.e., dns doesn't resolve ... maybe .. without knowing exactly why, I didn't want to be overly prescriptive.